packages/flutter/lib/src/services/platform_channel.dart

```diff
@@ -162,8 +238,7 @@ class BasicMessageChannel<T> {
 
   /// The messenger which sends the bytes for this channel, not null.
   BinaryMessenger get binaryMessenger {
```

Comment:
we should also update the API docs for all these classes to mention that they work on both the root isolate and background isolates, and what method to call to make sure they will work on the root isolate

Reply:
(my branch has an example of what that could look like but it'll need to be tweaked to the new semantics we discussed)

Reply:
Sorry, missed this, done.